processing/src/main/java/org/apache/druid/segment/ColumnarValueUnnestCursor.java

```diff
@@ -68,7 +70,10 @@ public ColumnSelectorFactory getColumnSelectorFactory()
       @Override
       public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec)
       {
-        return baseColumSelectorFactory.makeDimensionSelector(dimensionSpec);
+        if (!outputName.equals(dimensionSpec.getDimension())) {
+          return baseColumSelectorFactory.makeDimensionSelector(dimensionSpec);
+        }
+        return baseColumSelectorFactory.makeDimensionSelector(DefaultDimensionSpec.of(columnName));
```

Comment:
I'm perhaps missing something, but this seems to be just delegating to the base and returning without attempting to do any unnesting?

You likely haven't run into this because you are doing a validation ahead of time for what the column can be.  As such, the correct answer here might be to throw an UnsupportedOperatorException instead as you are expecting the user to be calling the ColumnValueSelector option instead.

Reply:
Addressed, added an unit test for the exception as well